dbms/src/Flash/Pipeline/Schedule/Events/Event.cpp

```diff
@@ -61,16 +68,22 @@ void Event::schedule() noexcept
     assert(0 == unfinished_inputs);
     exec_status.onEventSchedule();
     MemoryTrackerSetter setter{true, mem_tracker.get()};
-    // if err throw here, we should call finish directly.
-    bool direct_finish = true;
+    std::vector<TaskPtr> tasks;
     try
     {
-        // if no task is scheduled here, we can call finish directly.
-        direct_finish = scheduleImpl();
+        tasks = scheduleImpl();
+        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::random_pipeline_model_event_schedule_failpoint);
```

Comment:
What will happen when an expection throw in `scheudle() noexcept`

Reply:
For mocking exception thrown in scheduleImpl

Reply:
> For mocking exception thrown in scheduleImpl

But it will call std::terminate when exception thrown in function that mark as noexcept

Reply:
> But it will call std::terminate when exception thrown in function that mark as noexcept

Is it expected?

Reply:
the exception thrown in 'try catch', isn't it?